packages/domain-picker/src/domain-picker/index.tsx

```diff
@@ -71,6 +72,38 @@ export interface Props {
 	 * Name used to identify this component in tracks events.
 	 */
 	tracksName: string;
+	selectedDomain: DomainSuggestion | undefined;
+
+	/** The domain search query */
+	domainSearch: string;
+	/** Called when the domain search query is changed */
+	onSetDomainSearch: ( query: string ) => void;
+
+	/** The domain category */
+	domainCategory: string | undefined;
+
+	/** Called when the domain category is set */
+	onSetDomainCategory: ( category?: string ) => void;
+
+	/** Called when the modal is opened. Can be used for analytics */
+	onModalOpen: ( modalName: string ) => void;
+
+	/** Called when the modal is closed or unmounted in any way. Can be used for analytics */
+	// TODO: How to use analytics type here?
+	// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+	// @ts-ignore
```

Comment:
@sirreal any thoughts on this todo?

Reply:
We should know the shape of this function. We don't need the `RecordTrainTracksEventProps` type to type the function here. If the types align _structurally_ (do they have the same shape) they'll work. They don't need to align _nominally_ (come from the same definition).

I don't mind the `ts-ignore` being here in the first pass, but I would like them to go away _quickly_ 🙂

Reply:
Structurally... nominally... like that. Using plain ol' object for analytics event props. https://github.com/Automattic/wp-calypso/pull/42409/commits/2e4ae7ee2c74e00449cb75b1ee1da53181136d14